core/sched.cc

```diff
@@ -920,10 +948,6 @@ unsigned thread::realtime_priority() const
 
 void thread::set_realtime_time_slice(thread_realtime::duration time_slice)
 {
-    if (time_slice > 0) {
-        WARN_ONCE("set_realtime_time_slice() used but real-time time slices"
-                " not yet supported");
-    }
     _realtime._time_slice = time_slice;
```

Comment:
If it should be possible to change the time slice while the thread is already running, then we probably need to set the preemption timer here as well. Presently, I just assume that the slice is set before the thread is started.

Reply:
What if the new `_time_slice` is short enough that it should end its running slice now? Or maybe it means it should run until `now()` + new `_time_slice`?

Reply:
I think it is fine for set_realtime_time_slice() to not have an affect on a thread that is currently running.

Note that POSIX doesn't even have (!) a way to configure the realtime slice - if you use SCHED_RR it always uses some default time slice and you can't even change it, so this question doesn't even come up.

The worst case is when you switch from slice=0 to slice>0 - i.e., switch from SCHED_FIFO to SCHED_RR. I think it is fine for the change not to take affect until the thread yields or gets preempted for the first time. You can also add a comment stating that this is the case, and maybe we want to reconsider it in the future.

Reply:
Ok, thanks for the input. I added a short comment to the the function prototype in the sched.hh header file in 25357218742054b97f62c37844c6381fb70790f0